src/app/features/courses/courses.component.ts

```diff
@@ -1,30 +1,57 @@
 import { Component, OnInit } from '@angular/core';
-import { mockedAuthorsList, mockedCoursesList } from '@shared/mocks/mocks';
 import { Course } from '@shared/models/course.model';
-import { mapAuthorIdsToNames } from '@app/utilities/author-utils';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { UserStoreService } from '@app/user/services/user-store.service';
+import { Router } from '@angular/router';
+import { ButtonText } from '@shared/models/button.model';
 
 @Component({
   selector: 'app-courses',
   templateUrl: './courses.component.html',
   styleUrls: ['./courses.component.scss'],
 })
 export class CoursesComponent implements OnInit {
-  courses: Course[] = mockedCoursesList;
-  mappedCourses: { course: Course; authorsNames: string[] }[] = [];
-  selectedCourse!: Course;
+  // mappedCourses: { course: Course; authorsNames: string[] }[] = [];
+  courses!: Course[];
+  selectedCourse!: Course | null;
+  courseForEdit!: Course;
```

Comment:
You sure this needs to be a class member?

Reply:
hm, i don't know why i added it, deleted